tests/test_data_samples_order.py

```diff
@@ -0,0 +1,97 @@
+import random
+import pytest
+import copy
+
+
+ALGO_SCRIPT = """
+import json
+import substratools as tools
+class TestAlgo(tools.Algo):
+    def train(self, X, y, models, rank):
+        data_sample_keys = [folder.split('/')[-1] for folder in X]
+        assert data_sample_keys == {data_sample_keys}, data_sample_keys
```

Comment:
What's the purpose of this check?

As far as I know, this is comparing a list and a set, this will never be True.

Reply:
It's not actually a set, it's an interpolated variable. It is replaced by the actual list of data sample keys in the tests themselves.

Reply:
Oh got it!
To avoid confusion what about:
- using template string (https://docs.python.org/3/library/string.html#template-strings),
- or, renaming the variable to `TEMPLATE_ALGO_SCRIPT`